index.js

```diff
@@ -87,6 +88,18 @@ Compare.prototype = {
     if (x < 0) x = 0;
     if (x > this._bounds.width) x = this._bounds.width;
     return x;
+  },
+
+  setSlider: function(x) {
+    this._setPosition(x);
+  },
+
+  on: function(event, func) {
```

Comment:
I think you'll want to use a different pattern to emit events. [Here's a fairly simple example](https://github.com/tristen/pencil/blob/331dd2fbba6b165670f3085a931de5786cff2ca3/index.js) that uses `EventEmitter` with `on`, `off`, & `fire` methods to control event behaviour. So when a [line like this](https://github.com/mapbox/mapbox-gl-compare/blob/4e54668478436bd9fccaa48cfaf737e833ef39e1/index.js#L75-L78) is called, you would call something like

```js
this.fire('slideend', { currentPosition: this.currentPosition });
```

Reply:
Updated in 76704ad